src/sage/combinat/recognizable_series.py

```diff
@@ -456,7 +457,7 @@ def immutable(m):
             return m
 
         if isinstance(mu, dict):
-            mu = dict((a, immutable(M)) for a, M in mu.items())
+            mu = dict((a, immutable(Matrix(M))) for a, M in mu.items())
```

Comment:
```suggestion
            mu = dict((a, Matrix(M, immutable=True)) for a, M in mu.items())
```

Reply:
Indeed, that is the change I am suggesting. Although pushing for a bit more, I prefer the `dict` comprehension...:
```suggestion
            mu = {a: Matrix(M, immutable=True) for a, M in mu.items()}
```

Reply:
Thank you; needed to test something first, so didn't apply my original change instantanously. I also prefer the dict comprehension; so I applied.

Ideally, one should do the same for the vectors, but there seems to be a bug with immutable in that case, see #35693.